lang/java/avro/src/test/java/org/apache/avro/reflect/TestReflectData.java

```diff
@@ -90,4 +99,88 @@ public void testGenericProtocol() {
     private String bar;
     private int baz;
   }
+
+  protected static class DefaultReflector extends ReflectData {
+    private static final DefaultReflector INSTANCE = new DefaultReflector();
+
+    /** Return the singleton instance. */
+    public static DefaultReflector get() {
+      return INSTANCE;
+    }
+
+    private final Map<String, Object> defaultValues = new ConcurrentHashMap<>();
+
+    protected Object getOrCreateDefaultValue(String className) {
+      return this.defaultValues.computeIfAbsent(className, ignored -> {
+        try {
+          Class<?> aClass = Class.forName(className);
+          Constructor constructor = aClass.getDeclaredConstructor();
+          constructor.setAccessible(true);
+          return constructor.newInstance();
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException | NoSuchMethodException
+            | InvocationTargetException e) {
+          e.printStackTrace();
+        }
+        return null;
+      });
+    }
+
+    @Override
+    protected Object createSchemaDefaultValue(Type type, Field field, Schema fieldSchema) {
+      String className = ((Class) type).getName();
+      field.setAccessible(true);
+      Object def = null;
+
+      try {
+        Object value = getOrCreateDefaultValue(className);
+        if (value != null) {
+          def = field.get(value);
+        }
+      } catch (IllegalAccessException e) {
+        e.printStackTrace();
+      }
+
+      if (def == null) {
+        def = super.createSchemaDefaultValue(type, field, fieldSchema);
+      }
+
+      return def;
+    }
+  }
+
+  static class User {
+    public String first = "Avro";
+    public String last = "Apache";
+  }
+
+  static class Meta {
+    public int f1 = 55;
+    public String f2 = "a-string";
+    public List<String> f3 = Arrays.asList("one", "two", "three");
+    // public User usr = new User();
```

Comment:
@cutting I commented out this field due to an issue with JacksonUtils as described here https://issues.apache.org/jira/browse/AVRO-2775?